File: src/python/finished/counting_bits.py
from typing import List

# https://leetcode.com/problems/counting-bits/


class Solution:
    def countBits(self, n: int) -> List[int]:
        ans = [0] * (n + 1)
        for i in range(n + 1):
            ans[i] = (i & 1) + ans[i >> 1]
        return ans

File: src/rust/finished/counting_bits.rs
impl Solution {
    pub fn count_bits(n: i32) -> Vec<i32> {
        let n = n as usize;
        let mut ans = vec![0; n + 1];
        for i in 0..=n {
            ans[i] = (i & 1) as i32 + ans[i >> 1];
        }
        ans
    }
}
